tests/DataGovernanceTest.php

```diff
@@ -0,0 +1,229 @@
+        $responses = [
+            new Response(200, [], $body)
+        ];
+
+        $handler = $this->getHandlerStack($responses, $requests);
+        $fetcher = $this->getFetcher($handler);
+
+        // Act
+        $response = $fetcher->fetch("", "");
+
+        // Assert
+        $this->assertEquals(1, count($requests));
+        $this->assertContains($requests[0]['request']->getUri()->getHost(), ConfigFetcher::GLOBAL_URL);
+        $this->assertEquals(json_decode($body, true), $response->getBody());
+    }
+
+    public function testShouldStayOnSameUrlWithRedirect() {
+        // Arrange
+        $requests = [];
+        $body = sprintf(self::JSON_TEMPLATE, ConfigFetcher::GLOBAL_URL, 1);
+        $responses = [
+            new Response(200, [], $body)
+        ];
+
+        $handler = $this->getHandlerStack($responses, $requests);
+        $fetcher = $this->getFetcher($handler);
+
+        // Act
+        $response = $fetcher->fetch("", "");
+
+        // Assert
+        $this->assertEquals(1, count($requests));
+        $this->assertContains($requests[0]['request']->getUri()->getHost(), ConfigFetcher::GLOBAL_URL);
+        $this->assertEquals(json_decode($body, true), $response->getBody());
+    }
+
+    public function testShouldStayOnSameUrlEvenWhenForced() {
+        // Arrange
+        $requests = [];
+        $body = sprintf(self::JSON_TEMPLATE, ConfigFetcher::GLOBAL_URL, 2);
+        $responses = [
+            new Response(200, [], $body)
+        ];
+
+        $handler = $this->getHandlerStack($responses, $requests);
+        $fetcher = $this->getFetcher($handler);
+
+        // Act
+        $response = $fetcher->fetch("", "");
+
+        // Assert
+        $this->assertEquals(1, count($requests));
+        $this->assertContains($requests[0]['request']->getUri()->getHost(), ConfigFetcher::GLOBAL_URL);
+        $this->assertEquals(json_decode($body, true), $response->getBody());
+    }
+
+    public function testShouldRedirectToAnotherServer() {
+        // Arrange
+        $requests = [];
+        $firstBody = sprintf(self::JSON_TEMPLATE, ConfigFetcher::EU_ONLY_URL, 1);
+        $secondBody = sprintf(self::JSON_TEMPLATE, ConfigFetcher::EU_ONLY_URL, 0);
+        $responses = [
+            new Response(200, [], $firstBody),
+            new Response(200, [], $secondBody),
+        ];
+
+        $handler = $this->getHandlerStack($responses, $requests);
+        $fetcher = $this->getFetcher($handler);
+
+        // Act
+        $response = $fetcher->fetch("", "");
+
+        // Assert
+        $this->assertEquals(2, count($requests));
+        $this->assertContains($requests[0]['request']->getUri()->getHost(), ConfigFetcher::GLOBAL_URL);
+        $this->assertContains($requests[1]['request']->getUri()->getHost(), ConfigFetcher::EU_ONLY_URL);
+        $this->assertEquals(json_decode($secondBody, true), $response->getBody());
+    }
+
+    public function testShouldRedirectToAnotherServerWhenForced() {
+        // Arrange
```

Comment:
Currently, there is no big difference between this test and testShouldRedirectToAnotherServer().
In reality, the difference between the "forced" and "should" variants is that the "forced" variant is not allowed to use the "f" values of the first request.
Can we check that as well?